docs/develop/typescript/failure-detection.mdx

```diff
@@ -1,164 +1,152 @@
 ---
 id: failure-detection
-title: Failure detection
+title: Failure detection - TypeScript SDK feature guide
 sidebar_label: Failure detection
-slug: /develop/typescript/failure-detection
-toc_max_heading_level: 2
+description: Learn how to set Workflow and Activity Timeouts, implement Retry Policies, and manage Activity Heartbeats using the Temporal TypeScript SDK for efficient Workflow Execution.
+toc_max_heading_level: 4
 keywords:
-  - failure-detection
+  - guide context
+  - how to
+  - typescript
+  - sdk
+  - failure detection
 tags:
+  - guide-context
+  - how-to
+  - typescript
+  - sdk
   - failure-detection
-description: Learn about Workflow and Activity timeouts, retry policies, and heartbeat settings in Temporal. Optimize Workflow Execution by configuring WorkflowExecutionTimeout, WorkflowRunTimeout, and WorkflowTaskTimeout. Understand Activity timeout types like Schedule-To-Close, Start-To-Close, and Schedule-To-Start. Use Activity Heartbeats
 ---
 
-## Workflow timeouts {#workflow-timeouts}
+This page shows how to do the following:
+
+- [Workflow Timeouts](#workflow-timeouts)
+- [Workflow retries](#workflow-retries)
+- [Activity Timeouts](#activity-timeouts)
+- [Activity Retry Policy](#activity-retries)
+- [Activity Next Retry Delay](#activity-next-retry-delay)
+- [Heartbeat an Activity](#activity-heartbeats)
+- [Activity Heartbeat Timeout](#activity-heartbeat-timeout)
+
+## Workflow Timeouts {#workflow-timeouts}
+
+**How to set Workflow Timeouts using the Temporal TypeScript SDK**
 
 Each Workflow timeout controls the maximum duration of a different aspect of a Workflow Execution.
 
-Workflow timeouts are set when [starting the Workflow Execution](#workflow-timeouts).
+Workflow Timeouts are set when [starting the Workflow Execution](#workflow-timeouts).
 
 - **[Workflow Execution Timeout](/encyclopedia/detecting-workflow-failures#workflow-execution-timeout)** - restricts the maximum amount of time that a single Workflow Execution can be executed.
 - **[Workflow Run Timeout](/encyclopedia/detecting-workflow-failures#workflow-run-timeout):** restricts the maximum amount of time that a single Workflow Run can last.
 - **[Workflow Task Timeout](/encyclopedia/detecting-workflow-failures#workflow-task-timeout):** restricts the maximum amount of time that a Worker can execute a Workflow Task.
 
-Create an instance of `WorkflowOptions` from the Client and set your Workflow Timeout.
+The following properties can be set on the [`WorkflowOptions`](https://typescript.temporal.io/api/interfaces/client.WorkflowOptions/) when starting a Workflow using either the Client or Workflow API:
```

Comment:
Technically, isn't the Workflow API still acting as a client? Or am I missing something?

Reply:
No.

One can't use the client API from a Workflow, as the client API is not replace-aware and doing network requests could introduce non-deterministic behaviors.

Therefore, one has to use the Workflow API, which creates Workflow commands. This goes through a completely different path. And even though some operations are available both through the Client and the Workflow APIs, available options and the call semantic may differ.

Reply:
Oh, I get it now. I wasn't thinking of this from that angle. I thought you meant Workflow API meaning only "creating a new Workflow Execution", not "Using the `Workflow` type's API surface to create another Workflow" (Unless I've misunderstood again.)